app/models/grda_warehouse/sex_selector.rb

```
@@ -0,0 +1,143 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module GrdaWarehouse
+  # Determines the preferred Sex value from a set of source client records.
+  # This service object encapsulates the business rules for selecting the best
+  # available Sex value based on a multi-level sorting algorithm.
+  #
+  # The selection process prioritizes candidates based on the following criteria
+  # in order:
+  #   1. Sex value preference (0 or 1 preferred over 8, 9, 99)
+  #   2. Record timestamp (DateUpdated), preferring most recent
+  #   3. Record ID (lower is better)
+  #
+  # Valid Sex values:
+  #   0 => Female
+  #   1 => Male
+  #   8 => Client doesn't know
+  #   9 => Client prefers not to answer
+  #   99 => Data not collected
+  #
+  # This class handles data normalization for various source data formats
+  # (e.g., Hashes, ActiveRecord objects). The goal is to provide a consistent
+  # and reliable way to determine the most trustworthy Sex value from potentially
+  # conflicting source data.
+  class SexSelector
+    Candidate = Struct.new(:value, :tie_breakers, keyword_init: true)
+
+    def self.call(dest_attr:, source_clients:, use_oldest: false)
```

I wonder if instead of `use_oldest` we should accept a `prioritization_method:`.  Something like `prioritization_method: :newest_first`.  That way we could check for known methods, and raise if we get an unknown one, but then just call the method to return the sort order.  It could give us better flexibility long-term like "oldest except use DQ to prioritize" or "newest with a value" or something like that.

Also totally fine to leave this as is for now, we can revisit later, but since it's our first actual use of this, it feels like it might be worth making a bit more flexible.  For now we can just write the `:newest_first` method and raise if anything else is passed.

---

@eanders I've added updates to use a prioritization method instead of the boolean use_older. It looks like the PR auto requested a new Approval from you after the change too, but tagging you here in case it didnt send a notification.